### PR TITLE
Add snippet examples (#109)

### PR DESCRIPTION
**Pull request by sambergo**
It is quite well documented in [example config](https://github.com/L3MON4D3/LuaSnip/blob/master/Examples/snippets.lua), which is already linked (go straight to line 179). Not sure if this is too messy for docs. #109

### CODE REVIEW

File: docs/plugins/02-default-plugins.md [hunk 1]
```diff
@@ -106,6 +106,18 @@ View the documentation with `:help luasnip`
 
 LuaSnip uses friendly-snippets as it's snippet library.
 
+```lua
+-- Beside defining your own snippets you can also load snippets from "vscode-like" packages
+require("luasnip/loaders/from_vscode").load({ paths = { "./my-snippets" } })
```
Comment (kylo252) [inline docs/plugins/02-default-plugins.md:111]: I'm assuming this is unrelated to the lines after. Doesn't LuaSnip also looks for a `snippets` folder on your `runtimepath`? meaning, you can create `~/.config/lvim/snippets`.

File: docs/plugins/02-default-plugins.md [hunk 2]
```diff
@@ -106,6 +106,18 @@ View the documentation with `:help luasnip`
...
+-- Example on how to add your own
+local ls = require("luasnip")
+ls.snippets = {
+	javascript = {
+		ls.parser.parse_snippet("cv", 'console.log("${1:object} ", ${1})'),
+	},
+}
```
Comment (kylo252) [inline docs/plugins/02-default-plugins.md:118]: This should probably go higher, and specifically mention that this is the lua syntax.

**Review by kylo252 — CHANGES_REQUESTED** [review]
Thanks! we really need such a section, since a lot of people, including me, have no idea how to configure it.

File: docs/plugins/02-default-plugins.md (hunk 1 above)
Comment (sambergo) [reply docs/plugins/02-default-plugins.md:111]: Not 100% sure, but I don't think it does. You can store snippets wherever you want, so it can be shared folder with vscode for example. Just point line above to folder with package.json file, which will take care of the rest. Example:
https://github.com/dsznajder/vscode-es7-javascript-react-snippets/blob/2a6a1ffac598d7f5b4097d06c4190c5bcced99d9/package.json#L50

I'd assume this is what most people are looking for, or at least I was. 😅

Comment (kylo252) [reply docs/plugins/02-default-plugins.md:111]: @abzcoding, what's your take on this, since you might have some experience with it.

Comment (abzcoding) [reply docs/plugins/02-default-plugins.md:111]: i'm also doing this
```lua
require("luasnip/loaders/from_vscode").load { paths = { "~/.config/lvim/snippets" } }
```
and i have a `~/.config/lvim/snippets/package.json` which contains
```json
{
  "name": "nvim-snippets",
  "author": "abzcoding",
  "engines": {
    "vscode": "^1.11.0"
  },
  "contributes": {
    "snippets": [
      {
        "language": "lua",
        "path": "./lua.json"
      },
      {
        "language": [
          "typescript",
          "typescriptreact"
        ],
        "path": "./typescript.json"
      },
      {
        "language": "typescriptreact",
        "path": "./react-ts.json"
      }
    ]
  }
}
```
so i'm not sure my approach is the best one 😅

Comment (sambergo) [reply docs/plugins/02-default-plugins.md:111]: My approach is otherwise the same, but I keep my snippets separate from my dotfiles.

Comment (jw3126) [reply docs/plugins/02-default-plugins.md:111]: Thanks for this PR. Sadly this does not quite work for me. I added 

```lua
require("luasnip/loaders/from_vscode").load { paths = { "~/.config/lvim/mysnippets" } }
```
somewhere in my `config.lua` and created `"~/.config/lvim/mysnippets/package.json"` along with a snippet file. 
* However the snippets don't work and are not found by `:LuaSnipListAvailable`
* Also if I do
```lua
require("luasnip/loaders/from_vscode").load { paths = { "does not exist" } }
```
I don't get any error message. Am I missing something? Any idea how to debug this?

Comment (abzcoding) [reply docs/plugins/02-default-plugins.md:111]: this is what i get
Screen Shot 2022-01-03 at 4 48 11 PM

this is what i used 
```lua
  require("luasnip/loaders/from_vscode").load { paths = { "~/.config/lvim/snippets" } }
```
and here are my snippets -> https://github.com/abzcoding/lvim/tree/main/snippets

Comment (sambergo) [reply docs/plugins/02-default-plugins.md:111]: > 

I'ld try with existing snippet package. Clone https://github.com/dsznajder/vscode-es7-javascript-react-snippets for example and if that works you know problem is in your own package.json or other snippet files and you can mimic that package.

Comment (jw3126) [reply docs/plugins/02-default-plugins.md:111]: Thanks, @sambergo @abzcoding for the quick reply. The react snippets work, so it must be an issue with my snippets. Sorry for the noise (if somebody knows how to get an error message or other diagnostics on the snippets that would be awesome).

**Comment by sambergo** [comment]
I can't figure out a way to add examples to docs without it taking way too much space. I did https://github.com/sambergo/lunarvim-snippet-examples, which you could link to docs, edit or do anything that seems proper way.

File: docs/plugins/02-default-plugins.md (hunk 1 above)
Comment (sambergo) [reply docs/plugins/02-default-plugins.md:112]: > I'm assuming this is unrelated to the lines after. Doesn't LuaSnip also looks for a `snippets` folder on your `runtimepath`? meaning, you can create `~/.config/lvim/snippets`.

Yes it does. So if you have `~/.config/lvim/snippets/my-snippets/package.json` you can do both:
`require("luasnip/loaders/from_vscode").load { paths = { "~/.config/lvim/snippets/my-snippets" } }`
or
`require("luasnip/loaders/from_vscode").load { paths = { "my-snippets" } }`
